File: src/main/scala/application/compilerBuilder/ParticleList.scala
package application.compilerBuilder

import java.awt.event.MouseEvent

import core.particles.Particle
import org.jdesktop.swingx.JXList

class ParticleList() extends JXList() {
  override def getToolTipText(event: MouseEvent): String = {
    val index = this.locationToIndex(event.getPoint)
    val model = this.getModel
    if (index >= 0)
    {
      model.getElementAt(index).asInstanceOf[Particle].description
    }
    else
      ""
  }
}

File: playground/src/main/scala/miksilo/playground/application/compilerBuilder/ParticleList.java
package miksilo.playground.application.compilerBuilder;

import org.jdesktop.swingx.JXList;

import javax.swing.*;
import java.awt.event.MouseEvent;

public class ParticleList extends JXList {

    public String getToolTipText(MouseEvent event) {
        int index = this.locationToIndex(event.getPoint());
        ListModel model = this.getModel();
        if (index >= 0)
        {
            return DeltaInstance.DeltaLike(model.getElementAt(index)).getDelta().description();
        }
        else {
            return "";
        }
    }
}
